contracts/BatchExchangeViewer.sol

```diff
@@ -409,6 +409,12 @@ contract BatchExchangeViewer {
                 target,
                 offset + ADDRESS_WIDTH
             );
+        } else {
+            // NOTE: Ensure we write 0-s to the target memory location as we are
+            // reusing buffer and want to avoid stale data being left behind.
+            for (uint256 i = ADDRESS_WIDTH; i < AUCTION_ELEMENT_WIDTH; i++) {
```

Comment:
Sorry, I did a really crappy job reviewing this PR. We need to add the offset here, otherwise we are nulling the target buffer at the beginning but not necessarily at the right index.

🙈